### PR TITLE
feat(lax): add axis argument to `lax.top_k`

### LINKED ISSUE #25867: Add `axis` argument to `lax.top_k`
**Feature request:** Add an `axis` argument to [`lax.top_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.top_k.html).

Other similar reduction functions in [`lax`](https://jax.readthedocs.io/en/latest/jax.lax.html) receive an `axis` or `reduction_dimension` argument:
- [`argmax`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmax.html)
- [`argmin`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.argmin.html)
- [`approx_max_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_max_k.html)
- [`approx_min_k`](https://jax.readthedocs.io/en/latest/_autosummary/jax.lax.approx_min_k.html)

### PR DESCRIPTION
**Pull request by aymuos15**
Closes #25867

This is my first PR, please do let me know if I missed anything, thank you!

### CODE REVIEW

**Review by jakevdp — COMMENTED** [review]
Hi - thanks for the contribution!

The APIs in `jax.lax` are typically just small wrappers around XLA primitives like `top_k_p`. This change makes `top_k` a compound function that calls multiple primitives, which is not a suitable solution for code within `jax.lax`. In order to implement this, we'd probably want to add the `axis` argument to the `top_k_p` primitive itself, which is a bit tricky because the underlying chlo/stablehlo ops do not support the axis argument.

To be completely honest, this is probably not a good change to tackle for your first pull request in JAX.

**Comment by aymuos15** [comment]
Thank you very much for the feedback. I will close this for now then and can maybe approach it some other time with more experience.
